broker-core/src/main/java/io/zeebe/broker/logstreams/processor/TypedCommandWriter.java

```diff
@@ -25,7 +25,7 @@
 /** Things that any actor can write to a partition. */
 public interface TypedCommandWriter {
 
-  void appendNewCommand(Intent intent, UnpackedObject value);
+  void appendNewCommand(long key, Intent intent, UnpackedObject value);
```

Comment:
Since all new commands should not have a key, it is no intuitive to pass `-1` as key (i.e. magic number).

Reply:
I agree but the proposal in issue #2047 suggests an explicit API where either a valid key or -1 is passed. The idea is to force the user to always think about what the key should be in his situation. But I'm also ok with rolling this back, like you said, it's not very intuitive